### DISCUSSION TITLE
post/homelab-journey-internet/

### DISCUSSION THREAD

**Discussion by giscus** [ecwu/ecwu.github.io#11] [Blog Post Discussion]
# post/homelab-journey-internet/

Homepage and Blog of ECWU

https://ecwuuuuu.com/post/homelab-journey-internet/

**Comment by kuangmeng** [comment]
楼主列举的这些镜像，都已经打通SSO了嘛？

**Reply by ecwu** [reply]
> 楼主列举的这些镜像，都已经打通 SSO 了嘛？

不完全，其中
- Miniflux，Gitea，Outline，Grafana 是软件支持 OAuth 登录的
- Drone 通过 Gitea 登陆
- 剩下的软件没有 OAuth 支持，但是有一些没有鉴权的可以通过 Authentik 配合反向代理使用 [forward authentication](https://goauthentik.io/docs/providers/proxy/forward_auth) 进行接入。对于部署在公网的服务如果没有 SSO 支持，我会选择使用 Cloudflare Zero Trust 接入我自己搭建的 SSO。如果服务是内网服务，没必要搞这么复杂。

**Reply by kuangmeng** [reply]
我在自己的NAS上搞了一整套日常相关的网站并实现了OAuth2登录，楼主可以看看，有兴趣可以一起搞啊。https://nashome.tk

**Reply by ecwu** [reply]
都试用了下，能看出花了不少功夫搭建和维护，访问也挺流畅的。

提个小建议：可以考虑将目前独立的 draw.io 替换为 Confluence 里面的 draw.io 插件，后者允许将文件画好的 diagram 存储在服务器上，个人感觉体验更好。

**Comment by A1ca7raz** [comment]
2024年了，看起来homarr已经支持登陆鉴权了，日后还会更新sso集成，不过flame已经好久没有更新了

**Reply by ecwu** [reply]
个人感觉 flame 主打简洁，功能上属于够用。肯定是比不上 Homarr 更丰富的组件与集成功能，根据自己的需求选择就好。

**Comment by Atlantis-Gura** [comment]
> 对于部署在公网的服务如果没有 SSO 支持，我会选择使用 Cloudflare Zero Trust 接入我自己搭建的 SSO。

博主您好，由于暂住地没有公网IP，我目前需要外网访问的平台在用 Cloudflare Zero Trust 的 Access，配合 github 和 google 的 OAuth 做 SSO。我想请问一下，您说的 Cloudflare Zero Trust 接入自建 SSO 相比 Cloudflare 提供的 SSO 服务有何优点？

**Reply by ecwu** [reply]
Cloudflare（Zero Trust）支持的是 [SSO/Idp 的集成](https://developers.cloudflare.com/cloudflare-one/identity/idp-integration/)。那么自建 SSO （如我自建的 Authentik）或者连接 GitHub/Google 登录其实都是接上了一个外部的 Idp。Cloudflare 也在文档里说 Idp 集成是方便组织将自己已有的鉴权服务集成进来；我这边其实并没有查询到 Cloudflare 提供的 SSO 服务，我猜测你指的是通过邮箱发送一次性密码的那种形式。

我是用自建 SSO 的动机是因为这是我已经搭建好的服务，而且管理用户比较方便；如果是 Google 或者 GitHub 应该要配置邮箱域名限制或者直接写死邮箱。但如果只是个人使用我觉得是直接用 Google 或者 GitHub 会更方便的。

**Reply by Atlantis-Gura** [reply]
抱歉是我没有描述清楚。CF Zero Trust 中有个服务叫 Cloudflare Access，它提供的 Web 验证面板就跟 Authentik 的面板长得差不多，支持外部 SSO 的接入。

至于你说的邮箱发送一次性密码应该是叫 OTP (One-time PIN)，这种验证形式 Authentik 和 Cloudflare Access 应该都是支持的。Cloudflare Access 配置外部 SSO 比如 Google / GitHub 邮箱是直接在 Zero Trust 仪表盘完成的，也挺方便。

比如我现在搭建好了 Authentik，并且有公网ip，我想使用 Cloudflare Zero Trust 接入自建 Authentik，该如何设置呢？

**Reply by ecwu** [reply]
我是参考[官方文档](https://developers.cloudflare.com/cloudflare-one/identity/idp-integration/generic-oidc/)进行设置的，通过标准OIDC的方式添加。

**Reply by Atlantis-Gura** [reply]
好的，谢谢！

**Comment by WowYiJiu** [comment]
博主您好，我正在使用 Cloudflare Zero Trust 接入Authelia，想问一下你的Authentik的clent secret在Cloudflare填写时有什么加密吗，我根据Authelia官方文档随机生成的clinet secret粘贴到Cloudflare验证时一直报的是clinet secret错误，我采用两边都是明文形式也一样

**Reply by ecwu** [reply]
没有加密的，后台提供给我的 app id 和 secret 都直接填在对应空里

**Reply by ecwu** [reply]
![image](https://github.com/user-attachments/assets/0ed53029-87e6-4659-9517-362dbff66363)

配置如图供你参考。

**Reply by WowYiJiu** [reply]
非常感谢，我已经放弃了，现在也用Authentic了，今天继续跟着您视频学习，Authentic有图形管理界面自带LDAP感觉比Authelia好用多了
